### DISCUSSION TITLE
Firmware compatible with Ender 5?

### DISCUSSION THREAD

**Discussion by cds333** [mriscoc/Ender3V2S1#282] [Q&A]
Is it possible to modify this for use on an ender 5 plus? Assuming I am using the proper screen (not the built in touchscreen). According to creality the 4.2.2 and 4.2.7 boards work with it. I can't find any other firmware for the e5p that comes close to this.

Thank you!

**Comment by mriscoc** [comment]
You need to have a compatible board and also a DWIN or DACAI display unit from the V2 or S1.

**Reply by cds333** [reply]
I have the correct screen and board.

**Comment by cds333** [comment]
What other changes would be necessary besides inverting the direction of the Z axis and the bed size in the firmware? (Ender 5 Plus)

**Reply by mriscoc** [reply]
Check: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Comment by I-wish-2** [comment]
I to have an E 5 Pro that I added and E3V2 display to as well as the 4.2.7 Mother Board.  In addition I added the CR Touch.  when I woke up the printer after flashing both the MB and Display I found homing each Axis by itself was the best method as you can stop any stepper issues with the power button.  My notes show that I reversed at least one axis at the MB by switching pins 1(has the Tag) and pin 2.   As you noted it was the Z axis.
If your slicer is Cura please note that this slicer looks for a left front home.  The E5 will be right rear home and so the bed in Cura needs to be rotated 180 degrees,  XYZ indicators to the right rear.  This will set you print in Cura for the correct direction on your E5 bed.
This firmware is perfect for the E5 Pro, E3 Pro and the E3V2 I have not tested or run it on any other printers but you will be happy for sure when you have it all set up.
D

**Comment by Arashrk82** [comment]
I am using this on my Ender 3 v2, it is absolutely amazing! I really would like to install it on my Ender 5 pro as well. I have a spare E3V2 display. I just don't know what things need to be changed to make the firmware work with Ender 5 as well. Any help would be appreciated. Thanks.

**Reply by I-wish-2** [reply]
I noted all of the changes that I needed to have the E5 Pro work with Miguel's firmware in my post above.  The Z stepper will need to have it's direction changed in either the firmware, reversing the first two pins in the Z stepper cable at the Motherboard or use a reversing cable at the stepper motor plug.  I did upgrade my E5 board to the 4.2.7 and the display is the E3S1 style that does not show the thumbnails of the print that is loaded.  Check the latest release of the Miguel's firmware under the display updates to see which one you have and therefore which display firmware you will need. 

You need to have a compatible board and also a DWIN or DACAI display unit from the V2 or S1.  see notes above
Check: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides
D

**Reply by Arashrk82** [reply]
Thank you mate. The Ender 5, X-axis homing position is on the right side whereas the Ender 3 is on the left side. How did you change the homing position?

**Reply by I-wish-2** [reply]
Greetings,
Since the firmware will drive the axis in a direction, it's set in the firmware, then look for the home switch to cycle.  I simply reversed pins one and two on the motherboard plug to reverse the direction of the motor.  Pin one has the yellow tag on it.  If not switch the two wires on the right side of the plug as it is plug in the motherboard.  It can also be switched in the firmware on one of the configuration files but I'm not sure how.
Cheers
D

**Reply by Arashrk82** [reply]
So you reversed both Z and X-axis? That is not very ideal for me, I will have to compile the firmware once I figure out which section reverses the Z axis and change the homing position.

**Reply by I-wish-2** [reply]
Yes I reversed both the Z & X axis.  I too thought I could get around it but remember your are using E3 V2 firmware, it homes X to the left and Z down for either the Home switch or a BL or CR Touch.  Relocating the Z home switch was an big issue for me and so I added a CR Touch.  Setting up the X axis home to the left side of the gantry will require some modifications to the Hot End Wheel Plate if you use the CR Touch and even if you don't the left gantry inside front wheel will have an impact point with the wheel plate as the hot end is not centered on the plate.  I had to modify the plate to much, and I have a tooling shop, so reversing the steppers was the lesser of the efforts to get Miguel's firmware to run.  Which worked great for a year then the E5 became the first Klipper / Mainsail conversion.
If you use Cura Slicer it's origin is front left when the build platform is displayed.  Remember to rotate the the build platform 180 degrees if you do decide to reverse the steppers.  This will keep you print job in the orientation you needed especially if you have a single side parts cooling fan.
I wish you great success with your conversion and keep in touch about how it turned out.
Cheers
D
